lib/spacesuit/session_service.ex

```diff
@@ -63,15 +31,13 @@ defmodule Spacesuit.SessionService do
       {:ok, _, _} ->
         result
 
-      {:error, type, code, error} ->
+      {:error, type, code, error} when is_binary(error) ->
```

Comment:
Adding the guard clause allows us to treat this as a separate case instead of nesting flow control.